CMakeLists.txt

```diff
@@ -791,7 +791,7 @@ hpx_check_for_cxx14_std_integer_sequence(
 hpx_check_for_cxx14_std_result_of_sfinae(
   DEFINITIONS HPX_HAVE_CXX14_STD_RESULT_OF_SFINAE)
 
-if(NOT MSVC)
+if(NOT WIN32)
```

Comment:
I'm not a specialist about this, but I would say MinGW has very few POSIX headers, but it has at least POSIX threads